### PR TITLE
snap-patch/netplan: do not write files for temporary connections

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
Fixes LP:#1998207.

### CODE REVIEW

File: snap-patch/networkmanager/0002-netplan-make-use-of-libnetplan-for-YAML-backend.patch [hunk 1]
```diff
@@ -274,12 +274,13 @@ index ad6f277ce..0af209f64 100644
...
      nm_auto_unref_keyfile GKeyFile *kf_file        = NULL;
-@@ -409,6 +414,113 @@ _internal_write_connection(NMConnection                   *connection,
+@@ -409,6 +414,114 @@ _internal_write_connection(NMConnection                   *connection,
      if (existing_path && !existing_path_read_only && !nm_streq(path, existing_path))
          unlink(existing_path);
  
 +    /* NETPLAN: write only non-temporary files to /etc/netplan/... */
-+    if (!is_volatile && !is_nm_generated && !is_external) {
++    if (!is_volatile && !is_nm_generated && !is_external &&
++ strstr(keyfile_dir, "/etc/NetworkManager/system-connections")) {
```
Comment (slyon) [inline snap-patch/networkmanager/0002-netplan-make-use-of-libnetplan-for-YAML-backend.patch]: Nitpick: whitespace (probably got mangled by Launchpad copy&paste)
```suggestion
+        strstr(keyfile_dir, "/etc/NetworkManager/system-connections")) {
```

File: tests/vpn/openvpn/task.yaml [hunk 2]
```diff
@@ -42,4 +44,10 @@ execute: |
   fi
 
   network-manager.nmcli c down default
+
+  # Check that tun0/tun1/tun2 temporary connections do not have a static config file
+  if grep 'name: "tun' /etc/netplan/*; then
```
Comment (slyon) [inline tests/vpn/openvpn/task.yaml]: Suggestion (non-blocking): Be a bit more flexible with checking for quoted or unquoted "tun*" names and check inside `*.yaml` files only.
```suggestion
  if egrep 'name: "?tun' /etc/netplan/*.yaml; then
```

**Review by slyon — APPROVED** [review]
Thank you, this LGTM! Looks like the spread test (GitHub action passed, too).

Just a few (non-blocking) suggestions inline.

I've also confirmed our autopkgtest is now passing, using this source package: https://git.launchpad.net/network-manager/log/?h=netplan/lunar-gu

I confirmed our (previously failing) autopkgtest passes now for the OpenVPN duplicate connections fix: https://paste.ubuntu.com/p/FWTVxbjzC6/ (especially the TestNetplan.test_openvpn_connection case)

The change already landed in the "Netplan Everywhere" PPA: https://launchpad.net/~canonical-foundations/+archive/ubuntu/networkmanager-netplan

**Comment by alfonsosanchezbeato** [comment]
@slyon thanks for the suggestions!

**Pull request merged**